### PR TITLE
feat: integrate the new call ring flow

### PR DESCRIPTION
**Pull request by oliverlaz**
### Overview

Implements the latest spec about Ring Calls: https://www.notion.so/stream-wiki/Push-Notifications-a623dd508757404fa35ac4355a0bc814

### CODE REVIEW

**Comment by khushal87** [comment]
I was testing on the React messenger clone and found out that on clicking the call reject button, the request fails with an error:
```
{"StatusCode":404,"code":16,"details":[],"duration":"0","message":"Request failed with error: \"Not Found\"","more_info":"https://getstream.io/chat/docs/api_errors_response"}
```

File: packages/react-sdk/src/components/PendingCallPanel/PendingCallPanel.tsx [hunk 1]
```diff
@@ -21,24 +21,43 @@ const CALLING_STATE_TO_LABEL: Record<CallingState, string> = {
...
-        .map(({ user }) => user)
-        .filter((u) => !!u) || []
-    : caller
-    ? [caller]
-    : [];
+  const membersToShow: UserResponse[] = (members || [])
+    .slice(0, totalMembersToShow - (includeSelf ? 1 : 0))
+    .map(({ user }) => user)
+    .filter((user) => user.id !== connectedUser?.id || includeSelf);
+  if (
```
Comment (MartinCupela) [inline packages/react-sdk/src/components/PendingCallPanel/PendingCallPanel.tsx:52]: This code is a bit hard to follow. Could it be simplified?

**Review by szuperaz — APPROVED** [review]

**Comment by oliverlaz** [comment]
> I was testing on the React messenger clone and found out that on clicking the call accept and reject button, the request fails with an error:
> 
> ```
> {"StatusCode":404,"code":16,"details":[],"duration":"0","message":"Request failed with error: \"Not Found\"","more_info":"https://getstream.io/chat/docs/api_errors_response"}
> ```

 hey @khushal87, I can't reproduce this issue on my side. Can you try again on your end?

File: packages/client/src/Call.ts [hunk 2]
```diff
@@ -368,12 +368,10 @@ export class Call {
       if (this.isCreatedByMe && !hasOtherParticipants) {
         // Signals other users that I have cancelled my call to them
         // before they accepted it.
-        // Causes the `call.ended` event to be emitted to all the call members.
-        await this.endCall();
+        await this.reject();
```
Comment (MartinCupela) [inline packages/client/src/Call.ts:371]: This will send reject event during an active call. Is that expected?

File: packages/react-sdk/src/components/PendingCallPanel/PendingCallPanel.tsx (hunk 1 above)
Comment (oliverlaz) [reply packages/react-sdk/src/components/PendingCallPanel/PendingCallPanel.tsx:52]: I'll give it one more round :)
